src/libs/antares/study/parts/common/cluster_list.cpp

```diff
@@ -22,80 +24,63 @@ using namespace Antares;
 template<class ClusterT>
 inline uint ClusterList<ClusterT>::size() const
 {
-    return (uint)cluster.size();
+    return (uint)clusters.size();
 }
 
 template<class ClusterT>
 inline bool ClusterList<ClusterT>::empty() const
 {
-    return cluster.empty();
+    return clusters.empty();
 }
 
 template<class ClusterT>
 typename ClusterList<ClusterT>::iterator ClusterList<ClusterT>::begin()
 {
-    return std::begin(cluster);
+    return std::begin(clusters);
 }
 
 template<class ClusterT>
 typename ClusterList<ClusterT>::const_iterator ClusterList<ClusterT>::begin() const
 {
-    return std::begin(cluster);
+    return std::begin(clusters);
 }
 
 template<class ClusterT>
 typename ClusterList<ClusterT>::iterator ClusterList<ClusterT>::end()
 {
-    return std::end(cluster);
+    return std::end(clusters);
 }
 
 template<class ClusterT>
 typename ClusterList<ClusterT>::const_iterator ClusterList<ClusterT>::end() const
 {
-    return std::end(cluster);
+    return std::end(clusters);
 }
 
 template<class ClusterT>
-const ClusterT* ClusterList<ClusterT>::find(const Data::ClusterName& id) const
+ClusterT* ClusterList<ClusterT>::find(const Data::ClusterName& id) const
 {
-    auto i = cluster.find(id);
-    return (i != cluster.end()) ? i->second.get() : nullptr;
-}
+    const auto& it = std::ranges::find_if(clusters, [&id](auto& c) { return c->id() == id; });
 
-template<class ClusterT>
-ClusterT* ClusterList<ClusterT>::find(const Data::ClusterName& id)
-{
-    auto i = cluster.find(id);
-    return (i != cluster.end()) ? i->second.get() : nullptr;
+    return (it != clusters.end()) ? it->get() : nullptr;
 }
 
 template<class ClusterT>
-typename std::shared_ptr<ClusterT> ClusterList<ClusterT>::detach(iterator i)
+const ClusterT* ClusterList<ClusterT>::find(const ClusterT* p) const
```

Comment:
Now that we simplified **find(...)**, why wouldn't we have : 
```C++
template<class ClusterT>
const ClusterT* ClusterList<ClusterT>::find(const ClusterT* p) const
{
    return find(p->id());
}
```
?

Reply:
I don't remember if I once tested if this function `find(const ClusterT* p)` is called anywhere.
I would test that, but honestly, with the duration and CPU a recompilation would take, I feel lazy.

Reply:
Made a quick search, `find(const ClusterT* p)` does not seem to called.
We should try to remove it and see what happens at compilation.